src/pybel/manager/cache_manager.py

```diff
@@ -655,23 +655,36 @@ def iterate_singleton_edges_from_network(self, network):
                 ne1.c.network_id == network.id,
                 ne2.c.edge_id == None
             ))
-            .subquery('singleton_edge_ids_for_network')
         )
-        singleton_edges_for_network = (
-            self.session.query(Edge)
-            .filter(Edge.id.in_(singleton_edge_ids_for_network))
-        )
-        return iter(singleton_edges_for_network)
+        return singleton_edge_ids_for_network
 
     def drop_network(self, network):
         """Drops a network, while also cleaning up any edges that are no longer part of any network.
 
         :type network: Network
         """
-        for edge in self.iterate_singleton_edges_from_network(network):
-            self.session.delete(edge)
+        # get the IDs of the edges that will be orphaned by deleting this network
+        edge_ids = [result.edge_id for result in self.query_singleton_edges_from_network(network)]
```

Comment:
This is the only non-SQL bit I don't love, but we need this list of IDs in line 682 below, *after* we've deleted the rows in the table it comes from in line 676. Is there a better way?

Reply:
oh, if you just pass the query directly, it doesn't work?